aloha-core/src/test/scala/com/eharmony/aloha/models/multilabel/MultilabelModelTest.scala

```diff
@@ -47,124 +63,336 @@ class MultilabelModelTest extends ModelSerializationTestHelper {
+    assertEquals(None, result.value)
+    assertEquals("java.lang.Exception: error", result.errorMsgs.head.split("\n").head)
   }
 
   @Test def testSubvalueSuccess(): Unit = {
-    // Test the happy path by calling model.apply.  Check the value, missing data, and error messages.
-
-    fail()
+    val scoreToReturn = 5d
+    val modelSuccess = modelNoFeatures.copy(
+      labelsInTrainingSet = sci.IndexedSeq[Label]("label1", "label2", "label3", "label4"),
+      labelsOfInterest    = Option(labelsOfInterestExtractor),
+      predictorProducer   = Lazy(ConstantPredictor[Label](scoreToReturn)),
+      featureFunctions    = sci.IndexedSeq.empty
+    )
+
+    val result = modelSuccess(Map("a" -> "b", "label1" -> "label1", "label2" -> "label2"))
+    assertEquals(Vector(), result.errorMsgs)
+    assertEquals(Set(), result.missingVarNames)
+    assertEquals(Option(Map("label1" -> scoreToReturn, "label2" -> scoreToReturn)), result.value)
   }
 
   @Test def testExceptionsThrownInFeatureFunctionsAreNotCaught(): Unit = {
-    // NOTE: This is by design.
-
-    fail()
+    // This is by design.
+
+    val exception = new Exception("error")
+    val featureFunctionThatThrows: GenAggFunc[Map[String, String], Iterable[(String, Double)]] =
+      GenFunc0("", _ => throw exception)
+
+    val modelWithFeatureFunctionThatThrows = modelNoFeatures.copy(
+      featureNames        = sci.IndexedSeq("throwing feature"),
+      featureFunctions    = Vector(featureFunctionThatThrows),
+      labelsInTrainingSet = sci.IndexedSeq[Label](""),
+      labelsOfInterest    = None
+    )
+
+    val result = Try(modelWithFeatureFunctionThatThrows(Map()))
+    result match {
+      case Success(_) => fail()
+      case Failure(ex) => assertEquals(exception, ex)
+    }
   }
 }
 
 object MultilabelModelTest {
-  // TODO: Use this label type and Auditor.
-
+  // Types
   private type Label = String
   private val Auditor = RootedTreeAuditor.noUpperBound[Map[Label, Double]]()
 
-// TODO: Access information returned in audited value by using the following functions:
-  //    val aud: RootedTree[Any, Map[Label, Double]] = ???
-  //    aud.modelId           // : ModelIdentity
-  //    aud.value             // : Option[Map[Label, Double]]  // Should be missing on failure.
-  //    aud.missingVarNames   // : Set[String]
-  //    aud.errorMsgs         // : Seq[String]
-  //    aud.prob              // : Option[Float]  (Shouldn't need this)
+  case class ConstantPredictor[K](prediction: Double = 0d) extends SparseMultiLabelPredictor[K] {
+    override def apply(featuresUnused: SparseFeatures,
+      labels: Labels[K],
+      indicesUnused: LabelIndices,
+      ldfUnused: SparseLabelDepFeatures): Try[Map[K, Double]] =
+        Try(labels.map(_ -> prediction).toMap)
+  }
+
+  case class Lazy[A](value: A) extends (() => A) {
+    override def apply(): A = value
+  }
+
+  // Common input
+  val labelsInTrainingSet: sci.IndexedSeq[Label] = sci.IndexedSeq[Label]("a", "b", "c")
```

Comment:
These constants [should be capitalized](http://docs.scala-lang.org/style/naming-conventions.html#constants-values-variable-and-methods).